_data-prepper/pipelines/configuration/processors/anomaly-detector.md

```diff
@@ -0,0 +1,74 @@
+
+## Configuration
+
+You can configure the anomaly detector processor by specifying a key and the options for the selected mode. You can use the following options to configure the anomaly detector processor.
+
+| Name | Required | Description |
+| :--- | :--- | :--- |
+| `keys` | Yes | A non-ordered `List<String>` that is used as input to the ML algorithm to detect anomalies in the values of the keys in the list. At least one key is required.
+| `mode` | Yes |  The ML algorithm (or model) used to detect anomalies. You must provide a mode. See [random_cut_forest mode](#random_cut_forest-mode).
+
+### Keys
+
+Keys that are used in the anomaly detector processor are present in the input event. For example, if the input event is `{"key1":value1, "key2":value2, "key3":value3}`, then any of the keys (such as `key1`, `key2`, `key3`) in that input event can be used as anomaly detector keys as long as their value (such as `value1`, `value2`, `value3`) is an integer or real number.
+
+### random_cut_forest mode
+
+The random cut forest (RCF) ML algorithm is an unsupervised algorithm for detecting anomalous data points within a dataset. To detect anomalies, the anomaly detector processor uses the `random_cut_forest` mode.
+
+| Name | Description |
+| :--- | :--- |
+| `random_cut_forest` | Processes events using the RCF ML algorithm to detect anomalies. After passing a group of events with latency, a value between 0.2 and 0.3 is passed through the anomaly detector processor. When an event with latency value 11.5 is sent, the following anomaly event is generated.| 
+
+The following example shows what happens in the anomaly detector processor when it receives input:
+
+ ```json
+  { "latency": 11.5, "deviation_from_expected":[10.469302736820003],"grade":1.0}
+```
+
+In this example, `deviation_from_expected` is a list of deviations for each of the keys from their corresponding expected values, and `grade` is the anomaly grade that indicates the anomaly severity.
+     
+
+You can configure `random_cut_forest` mode with the following options. 
+
+| Name | Default value | Range | Description |
+| :--- | :--- | :--- | :--- |
+| `shingle_size` | `4` | 1 - 60 | The shingle size used in the ML algorithm. |
+| `sample_size` | `256` | 100 - 2500 | The sample size size used in the ML algorithm. |
+| `time_decay` | `0.1` | 0 - 1.0 | The time decay value used in the ML algorithm. Used as the mathematical expression `timeDecay` divided by `SampleSize` in the ML algorithm. |
+| `type` | `metrics` | N/A | The type of data sent to the algorithm. |
+| `version` | `1.0` | N/A | The algorithm version number. |
+
+## Usage
+
+To get started, create the following `pipeline.yaml` file. You can use the following pipeline configuration to look for anomalies in the `latency` field in events that are passed to the processor. Then, you can use the following YAML configuration file `random_cut_forest` mode to detect anomalies:
```

Comment:
Skip the comma after single-word adverbs of time at the beginning of a sentence, such as afterward, then, later, or subsequently.